Error build

```
rpmbuild -ba raft.spec 
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.59132
+ umask 022
+ cd /home/builder/rpmbuild/BUILD
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ cd /home/builder/rpmbuild/BUILD
+ rm -rf raft-0.17.4
+ /bin/gzip -dc /home/builder/rpmbuild/SOURCES/raft-0.17.4.tar.gz
+ /usr/bin/tar -xf -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd raft-0.17.4
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.59132
+ umask 022
+ cd /home/builder/rpmbuild/BUILD
+ cd raft-0.17.4
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ autoreconf -i
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'ac'.
libtoolize: copying file 'ac/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:9: installing 'ac/compile'
configure.ac:11: installing 'ac/config.guess'
configure.ac:11: installing 'ac/config.sub'
configure.ac:7: installing 'ac/install-sh'
configure.ac:7: installing 'ac/missing'
Makefile.am: installing 'ac/depcomp'
parallel-tests: installing 'ac/test-driver'
+ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for UV... yes
checking for LZ4... yes
checking whether to build with code coverage support... no
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking assert.h usability... yes
checking assert.h presence... yes
checking for assert.h... yes
checking for unistd.h... (cached) yes
checking linux/io_uring.h usability... yes
checking linux/io_uring.h presence... yes
checking for linux/io_uring.h... yes
checking linux/aio_abi.h usability... yes
checking linux/aio_abi.h presence... yes
checking for linux/aio_abi.h... yes
checking whether RWF_NOWAIT is declared... yes
checking for zfs... no
checking for size_t... yes
checking for ssize_t... yes
checking for uint8_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking if gcc supports flag -std=c11 in envvar CFLAGS... yes
checking if gcc supports flag -g in envvar CFLAGS... yes
checking if gcc supports flag --mcet in envvar CFLAGS... no
checking if gcc supports flag -fcf-protection in envvar CFLAGS... no
checking if gcc supports flag --param=ssp-buffer-size=4 in envvar CFLAGS... yes
checking if gcc supports flag -pipe in envvar CFLAGS... yes
checking if gcc supports flag -fno-strict-aliasing in envvar CFLAGS... yes
checking if gcc supports flag -fdiagnostics-color in envvar CFLAGS... yes
checking if gcc supports flag -fexceptions in envvar CFLAGS... yes
checking if gcc supports flag -fstack-clash-protection in envvar CFLAGS... no
checking if gcc supports flag -fstack-protector-strong in envvar CFLAGS... yes
checking if gcc supports flag -fasynchronous-unwind-tables in envvar CFLAGS... yes
checking if gcc supports flag -fdiagnostics-show-option in envvar CFLAGS... yes
checking if gcc supports flag -Wall in envvar CFLAGS... yes
checking if gcc supports flag -Wextra in envvar CFLAGS... yes
checking if gcc supports flag -Wpedantic in envvar CFLAGS... yes
checking if gcc supports flag -Wimplicit-fallthrough=5 in envvar CFLAGS... no
checking if gcc supports flag -Wcast-align in envvar CFLAGS... yes
checking if gcc supports flag -Wstrict-prototypes in envvar CFLAGS... yes
checking if gcc supports flag -Wlogical-op in envvar CFLAGS... yes
checking if gcc supports flag -Wmissing-include-dirs in envvar CFLAGS... yes
checking if gcc supports flag -Wold-style-definition in envvar CFLAGS... yes
checking if gcc supports flag -Winit-self in envvar CFLAGS... yes
checking if gcc supports flag -Wfloat-equal in envvar CFLAGS... yes
checking if gcc supports flag -Wsuggest-attribute=noreturn in envvar CFLAGS... yes
checking if gcc supports flag -Wformat=2 in envvar CFLAGS... yes
checking if gcc supports flag -Wendif-labels in envvar CFLAGS... yes
checking if gcc supports flag -Wdate-time in envvar CFLAGS... yes
checking if gcc supports flag -Wnested-externs in envvar CFLAGS... yes
checking if gcc supports flag -Wconversion in envvar CFLAGS... yes
checking if gcc supports flag -Werror=implicit-function-declaration in envvar CFLAGS... yes
checking if gcc supports flag -Wunused-but-set-variable in envvar CFLAGS... yes
checking if gcc supports flag -Werror=return-type in envvar CFLAGS... yes
checking if gcc supports flag -Werror=incompatible-pointer-types in envvar CFLAGS... yes
checking if gcc supports flag -Wshadow in envvar CFLAGS... yes
checking if gcc supports flag -Werror=overflow in envvar CFLAGS... yes
checking if gcc supports flag -Werror=shift-count-overflow in envvar CFLAGS... yes
checking if gcc supports flag -Werror=shift-overflow=2 in envvar CFLAGS... no
checking if gcc supports flag -Warray-bounds in envvar CFLAGS... yes
checking if gcc supports flag -Wrestrict in envvar CFLAGS... no
checking if gcc supports flag -Wreturn-local-addr in envvar CFLAGS... yes
checking if gcc supports flag -Wstringop-overflow in envvar CFLAGS... no
checking if gcc supports flag -z in envvar LDFLAGS... no
checking if gcc supports flag relro in envvar LDFLAGS... no
checking if gcc supports flag -z in envvar LDFLAGS... (cached) no
checking if gcc supports flag now in envvar LDFLAGS... no
checking if gcc supports flag -fstack-protector-strong in envvar LDFLAGS... yes
checking if gcc supports flag --param=ssp-buffer-size=4 in envvar LDFLAGS... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating raft.pc
config.status: creating Makefile
config.status: executing depfiles commands
config.status: executing libtool commands
+ /usr/bin/make -j8
  CC       src/libraft_la-byte.lo
  CC       src/libraft_la-client.lo
  CC       src/libraft_la-compress.lo
  CC       src/libraft_la-configuration.lo
  CC       src/libraft_la-convert.lo
  CC       src/libraft_la-election.lo
  CC       src/libraft_la-entry.lo
  CC       src/libraft_la-err.lo
src/err.c: In function 'errMsgWrap':
src/err.c:30:9: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wformat-truncation"
         ^
  CC       src/libraft_la-flags.lo
  CC       src/libraft_la-heap.lo
  CC       src/libraft_la-legacy.lo
  CC       src/libraft_la-lifecycle.lo
  CC       src/libraft_la-log.lo
  CC       src/libraft_la-membership.lo
  CC       src/libraft_la-progress.lo
  CC       src/libraft_la-random.lo
  CC       src/libraft_la-raft.lo
  CC       src/libraft_la-recv.lo
  CC       src/libraft_la-recv_append_entries.lo
  CC       src/libraft_la-recv_append_entries_result.lo
  CC       src/libraft_la-recv_request_vote.lo
  CC       src/libraft_la-recv_request_vote_result.lo
  CC       src/libraft_la-recv_install_snapshot.lo
  CC       src/libraft_la-recv_timeout_now.lo
  CC       src/libraft_la-replication.lo
  CC       src/libraft_la-snapshot.lo
  CC       src/libraft_la-start.lo
  CC       src/libraft_la-state.lo
  CC       src/libraft_la-syscall.lo
  CC       src/libraft_la-task.lo
  CC       src/libraft_la-tick.lo
  CC       src/libraft_la-tracing.lo
  CC       src/libraft_la-fixture.lo
  CC       src/libraft_la-uv.lo
  CC       src/libraft_la-uv_append.lo
  CC       src/libraft_la-uv_encoding.lo
  CC       src/libraft_la-uv_finalize.lo
  CC       src/libraft_la-uv_fs.lo
  CC       src/libraft_la-uv_ip.lo
  CC       src/libraft_la-uv_list.lo
src/uv.c:6:24: fatal error: sys/random.h: No such file or directory
compilation terminated.
make: *** [Makefile:2250: src/libraft_la-uv.lo] Error 1
make: *** Waiting for unfinished jobs....
error: Bad exit status from /var/tmp/rpm-tmp.59132 (%build)
```

```
rpm -qa | grep uv
lib64drm_nouveau2-2.4.103-1-rosa2016.1.x86_64
lib64uv1-1.44.2-1-rosa2016.1.x86_64
lib64uv-devel-1.44.2-1-rosa2016.1.x86_64
```

```
uname -a
Linux 1f450d2ed088 5.13.9-generic-1rosa-x86_64 #1 SMP PREEMPT Sun Aug 8 20:12:26 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

```
lsb_release -a
LSB Version:    *
Distributor ID: RosaDesktopFresh
Description:    ROSA Desktop Fresh R11.1
Release:        2016.1
Codename:       Fresh
```

--- a/src/uv.c
+++ b/src/uv.c
@@ -3,7 +3,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/random.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -609,20 +608,19 @@ static int uvRandom(struct raft_io *io, int min, int max)
 
 static void uvSeedRand(struct uv *uv)
 {
-    ssize_t sz = -1;
-    unsigned seed = 0; /* fed to srand() */
+    struct timeval time = {0};
+    unsigned seed = 0;
 
-    sz = getrandom(&seed, sizeof seed, GRND_NONBLOCK);
-    if (sz == -1 || sz < ((ssize_t)sizeof seed)) {
-        /* Fall back to an inferior random seed when `getrandom` would have
-         * blocked or when not enough randomness was returned. */
-        seed ^= (unsigned)uv->id;
-        seed ^= (unsigned)uv_now(uv->loop);
-        struct timeval time = {0};
-        /* Ignore errors. */
-        gettimeofday(&time, NULL);
-        seed ^= (unsigned)((time.tv_sec * 1000) + (time.tv_usec / 1000));
-    }
+    /* Create a seed by combining the node ID and the current time.
+     *
+     * Given that we use rand() only to generate the randomized election timeout
+     * and not to, say, perform some secure-sensitive cryptographic task, it's
+     * ok to not use more sophisticated methods to create the seed (for
+     * example. getrandom()). */
+    seed ^= (unsigned)uv->id;
+    seed ^= (unsigned)uv_now(uv->loop);
+    gettimeofday(&time, NULL);
+    seed ^= (unsigned)((time.tv_sec * 1000) + (time.tv_usec / 1000));
 
     srand(seed);
 }

